sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -236,6 +238,52 @@ public <T> T insert(String containerName, T objectToSave, PartitionKey partition
         return toDomainObject(domainType, response.getItem());
     }
 
+    /**
+     * applies partial update (patch) to an item with CosmosPatchItemRequestOptions
+     *
+     * @param entityToPatch must not be {@literal null}, must contain {@link org.springframework.data.annotation.Id}
+     * and {@link com.azure.spring.data.cosmos.core.mapping.PartitionKey} field
+     * @param patchOperations must not be {@literal null}
+     * @param options optional CosmosPatchItemRequestOptions, e.g. options.setFilterPredicate("FROM products p WHERE p.used = false");
+     * @param <T> type class of domain type
+     * @return the inserted item
+     */
+    public <T> T patch(T entityToPatch, CosmosPatchOperations patchOperations, CosmosPatchItemRequestOptions options) {
+        Assert.notNull(patchOperations, "expected non-null cosmosPatchOperations");
+
+        @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) entityToPatch.getClass();
+
+        final String containerName = getContainerName(entityToPatch.getClass());
+        LOGGER.debug("execute patchItem in database {} container {}", this.getDatabaseName(),
```

Comment:
maybe add check to only log if `LOGGER.isDebugEnabled()`

Reply:
@kushagraThapar is this necessary?

Reply:
Would be a good practice to do. However, since there are no computations in this log line, we don't really need it need it.